build/vsts_build.yaml

```diff
@@ -340,14 +347,46 @@ phases:
       scriptName: "$(Build.Repository.LocalPath)\\scripts\\cibuild\\BuildValidator.ps1"
       arguments: "-BuildRTM $(BuildRTM) -BuildOutputTargetPath $(BuildOutputTargetPath) -ValidateVsix"
     condition: " and(succeeded(), eq(variables['IsOfficialBuild'], 'true')) "
-  
+
   - task: PowerShell@1
     displayName: "Validate Repository Artifacts Localization"
     inputs:
       scriptName: "$(Build.Repository.LocalPath)\\scripts\\cibuild\\BuildValidator.ps1"
       arguments: "-BuildRTM $(BuildRTM) -BuildOutputTargetPath $(BuildOutputTargetPath)"
     condition: " and(succeeded(), eq(variables['IsOfficialBuild'], 'true')) "
 
+    # Use dotnet msbuild instead of MSBuild CLI.
+    # Using MSBuild CLI results in an assembly binding failure for NuGet.Common.dll 4.4.0.3 because Microsoft.DotNet.Build.Tasks.Feed.dll references SleetLib.dll which references NuGet.Common 4.4.0.3.
+    # The Microsoft.DotNet.Build.Tasks.Feed package includes NuGet.Common 4.9.0.6 and a binding redirection in Microsoft.DotNet.Build.Tasks.Feed.dll.config but the binding redirection is not processed.
+    # This would probably solve it:  https://github.com/Microsoft/msbuild/issues/1309
+  - task: CmdLine@2
+    displayName: "Publish the packages to the .NET Core build asset registry (BAR)"
+    inputs:
+      script: dotnet msbuild ..\build\publish.proj /t:PublishPackagesToBuildAssetRegistry /p:NuGetClientNupkgsDirectoryPath=$(Build.Repository.LocalPath)\artifacts\$(NupkgOutputDir) /p:FeedUrl=$(DotNetFeedUrl) /p:BuildId=$(Build.BuildNumber) /p:RepoBranch=$(Build.SourceBranchName) /p:RepoCommit=$(Build.SourceVersion) /p:RepoUri=$(Build.Repository.Uri) /p:ManifestFilePath=$(ManifestFilePath) /p:AccountKey=$(dotnetfeed-storage-access-key-1)
```

Comment:
>DotNetFeedUrl [](start = 195, length = 13)

Where is this defined?

Reply:
It's defined as a variable on our Azure DevOps build definition.

Reply:
Got it!

---
In reply to: [272359675](https://github.com/NuGet/NuGet.Client/pull/2794#discussion_r272359675) [](ancestors = 272359675)